pyproject.toml

```diff
@@ -48,3 +48,2469 @@ filterwarnings = [
+    "sentry.rules.actions.integrations.create_ticket.form",
+    "sentry.rules.actions.integrations.create_ticket.utils",
+    "sentry.rules.actions.notify_event_service",
+    "sentry.rules.actions.sentry_apps.notify_event",
+    "sentry.rules.conditions.event_attribute",
+    "sentry.rules.conditions.event_frequency",
+    "sentry.rules.conditions.level",
+    "sentry.rules.conditions.tagged_event",
+    "sentry.rules.filters.age_comparison",
+    "sentry.rules.filters.assigned_to",
+    "sentry.rules.filters.issue_category",
+    "sentry.rules.filters.issue_occurrences",
+    "sentry.rules.filters.latest_release",
+    "sentry.rules.history.backends.postgres",
+    "sentry.rules.history.endpoints.project_rule_group_history",
+    "sentry.rules.history.endpoints.project_rule_stats",
+    "sentry.rules.history.preview",
+    "sentry.rules.processor",
+    "sentry.rules.registry",
+    "sentry.runner.commands.backup",
+    "sentry.runner.commands.cleanup",
+    "sentry.runner.commands.createuser",
+    "sentry.runner.commands.devserver",
+    "sentry.runner.commands.exec",
+    "sentry.runner.commands.files",
+    "sentry.runner.commands.migrations",
+    "sentry.runner.commands.permissions",
+    "sentry.runner.commands.plugins",
+    "sentry.runner.commands.run",
+    "sentry.runner.commands.spans",
+    "sentry.runner.commands.tsdb",
+    "sentry.runner.commands.upgrade",
+    "sentry.runner.importer",
+    "sentry.runner.initializer",
+    "sentry.runner.settings",
+    "sentry.scim.endpoints.members",
+    "sentry.scim.endpoints.schemas",
+    "sentry.scim.endpoints.teams",
+    "sentry.scim.endpoints.utils",
+    "sentry.sdk_updates",
+    "sentry.search.events.builder.discover",
+    "sentry.search.events.builder.issue_platform",
+    "sentry.search.events.builder.metrics",
+    "sentry.search.events.builder.spans_metrics",
+    "sentry.search.events.constants",
+    "sentry.search.events.datasets.base",
+    "sentry.search.events.datasets.discover",
+    "sentry.search.events.datasets.filter_aliases",
+    "sentry.search.events.datasets.function_aliases",
+    "sentry.search.events.datasets.metrics",
+    "sentry.search.events.datasets.metrics_layer",
+    "sentry.search.events.datasets.profile_functions",
+    "sentry.search.events.datasets.profiles",
+    "sentry.search.events.datasets.spans_metrics",
+    "sentry.search.events.fields",
+    "sentry.search.events.filter",
+    "sentry.search.snuba.backend",
+    "sentry.search.snuba.executors",
+    "sentry.search.utils",
+    "sentry.security.emails",
+    "sentry.security.utils",
+    "sentry.sentry_apps.apps",
+    "sentry.sentry_apps.components",
+    "sentry.sentry_apps.installations",
+    "sentry.sentry_metrics.consumers.indexer.slicing_router",
+    "sentry.sentry_metrics.consumers.last_seen_updater",
+    "sentry.sentry_metrics.indexer.postgres.apps",
+    "sentry.sentry_metrics.indexer.postgres.models",
+    "sentry.sentry_metrics.indexer.postgres.postgres_v2",
```

Comment:
fwiw a few of these will be fixed in a followup automatically and opted in automatically -- but the rest are real errors that'll need fixing:

```console
$ make backend-typing
--> Running Python typing checks
mypy
src/sentry/sentry_metrics/indexer/postgres/apps.py:4: error: Unused "type: ignore" comment
src/sentry/sentry_metrics/indexer/postgres/models.py:19: error: Unused "type: ignore" comment
src/sentry/sentry_metrics/indexer/postgres/models.py:22: error: Need type annotation for "string"  [var-annotated]
src/sentry/sentry_metrics/indexer/postgres/models.py:23: error: Need type annotation for "date_added"  [var-annotated]
src/sentry/sentry_metrics/indexer/postgres/models.py:25: error: Unused "type: ignore" comment
src/sentry/sentry_metrics/indexer/postgres/models.py:45: error: Unused "type: ignore" comment
src/sentry/sentry_metrics/indexer/postgres/models.py:46: error: Need type annotation for "string"  [var-annotated]
src/sentry/sentry_metrics/indexer/postgres/models.py:48: error: Need type annotation for "date_added"  [var-annotated]
src/sentry/sentry_metrics/indexer/postgres/models.py:49: error: Need type annotation for "last_seen"  [var-annotated]
src/sentry/sentry_metrics/indexer/postgres/models.py:50: error: Need type annotation for "retention_days"  [var-annotated]
src/sentry/sentry_metrics/indexer/postgres/models.py:52: error: Unused "type: ignore" comment
src/sentry/sentry_metrics/indexer/postgres/models.py:73: error: Need type annotation for "use_case_id"  [var-annotated]
src/sentry/sentry_metrics/consumers/indexer/slicing_router.py:148: error: Argument "sliceable" to "SlicingRouter" has incompatible type "str"; expected "Literal['generic_metrics']"  [arg-type]
src/sentry/sentry_metrics/indexer/postgres/postgres_v2.py:408: error: M? has no attribute "organization_id"  [attr-defined]
src/sentry/sentry_metrics/indexer/postgres/postgres_v2.py:409: error: M? has no attribute "string"  [attr-defined]
src/sentry/sentry_metrics/consumers/last_seen_updater.py:31: error: Unused "type: ignore" comment
Found 16 errors in 5 files (checked 4734 source files)
make: *** [Makefile:156: backend-typing] Error 1
```

Reply:
> it already wasn't being caught because of --strict being a lie

I don't really know much about --strict, but basically, if I do something obviously broken like `return "foo" + 42` in any of those now-ignored files, `make backend-typing` on master catches it, but after this PR it won't.

I concretely applied this patch on both master and this PR to compare how `make backend-typing` behaves:

```diff
diff --git a/src/sentry/sentry_metrics/consumers/last_seen_updater.py b/src/sentry/sentry_metrics/consumers/last_seen_updater.py
index bd559fa4fa..1a4f2657bb 100644
--- a/src/sentry/sentry_metrics/consumers/last_seen_updater.py
+++ b/src/sentry/sentry_metrics/consumers/last_seen_updater.py
@@ -184,3 +184,7 @@ def get_last_seen_updater(
         processing_factory,
         IMMEDIATE,
     )
+
+
+def foo(x: str):
+    return x + 1
```

I don't think it's a big deal to be clear and I certainly wouldn't want to block this PR from landing because of things like this

Reply:
yes that is correct -- but also there are lots of hidden bugs in our existing code because we're _trusting_ strict and it is lying to us.  it's a trade off and this patch is basically the only sane way out of this mess